Feature/connector

amitmahindrakar opened the pull request.

amitmahindrakar committed:
Merge branch 'scratch' of github.com:DemocracyDevelopers/colorado-rla into feature/connector

amitmahindrakar committed:
Merge branch 'scratch' of github.com:DemocracyDevelopers/colorado-rla into feature/connector

amitmahindrakar committed:
added code to retreive results by contestname

amitmahindrakar committed:
added code to retreive results by contestname

amitmahindrakar commented on server/eclipse-project/pom.xml:
```diff
@@ -319,11 +319,6 @@
 			<artifactId>maven-site-plugin</artifactId>
 			<version>3.7.1</version>
 		</dependency>
-		<dependency>
```
This is not required.

amitmahindrakar commented on server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/GenerateAssertions.java:
```diff
@@ -116,33 +130,64 @@ public String endpointBody(final Request the_request, final Response the_respons
                     .build()));
+
+    // Following is temporary code until we integrate and get real data for contest names
+    //TODO start: remove
```
This is test code until we get a list of contestNames  from request.

amitmahindrakar commented on server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/GenerateAssertions.java:
```diff
@@ -116,33 +130,64 @@ public String endpointBody(final Request the_request, final Response the_respons
+    List<AuditResponse> raireResponse = objectMapper.convertValue(generic, new TypeReference<List<AuditResponse>>() { });
+
+
+    raireResponse.stream().forEach(auditResponse -> {
```
This can do with a bit of cleanup. I especially don't like line number 171.

vteague replied:
And on line 182 we'll need the new NEN assertion to have the relevant data, e.g. winner, loser, continuing candidates, analogous to new NEBAssertion in ln 177.

The universeSize is the whole collection of ballots being audited in this 'universe', which is either the whole county for county-wide contests, or the whole state for statewide contests. I'll explain how to find it.

vteague committed:
Add EstimateSampleSizes to endpoint_classes.

vteague committed:
Merge pull request #52 from DemocracyDevelopers/IRVBallotUtils

Add EstimateSampleSizes to endpoint_classes.

vteague committed:
Add EstimateSampleSizes to endpoint_classes.

vteague committed:
Merge pull request #53 from DemocracyDevelopers/IRVBallotUtils

Added exhaustive test class

vteague committed:
Started sketching IRV ballot interpretation at the CSV parse point.

vteague committed:
Merge pull request #54 from DemocracyDevelopers/scratch

Scratch

vteague committed:
Added some code and tests for IRV ballot interpretation as a list of (candidate name) strings.

vteague committed:
Update choice-parsing code so that it thinks the choices are "Alice", "Bob" not "Alice(1)" "Alice(2)" "Bob(1)" "Bob(2)" etc.

vteague committed:
Updating contest choices seems to be working. CVRs not yet updated.

vteague committed:
Working on updating CVRs with new votes (without explicit ranks in parentheses).

vteague committed:
IRV parsing now working for complicated candidate names.

vteague committed:
Removed IRVCountyContestResult.java

michelleblom committed:
Added method for removing discrepancies in Assertion classes; set database to scratch in property files

michelleblom committed:
Merge branch 'scratch' of https://github.com/DemocracyDevelopers/colorado-rla into scratch

vteague committed:
Cleaner IRV contest rewriting

vteague committed:
Cleaner IRV contest rewriting

vteague committed:
Cleaned up default CountyContestResult values for IRV.

vteague committed:
IRV CSV parsing working. Still need to check details for write-ins. Error messages are not necessarily particularly helpful.

vteague committed:
Reorganized some IRV classes in model.

vteague committed:
Comments about some better ways to do it.

vteague committed:
Altered it so that the _contest_ choices remain as "Alice (1)", "Alice (2)" etc, but the _CVR_ choices are an ordered list of plain names.

michelleblom committed:
Revised sample size estimation to produce list of string arrays for eventual export to a CSV file.

michelleblom committed:
Replaced the annotations of the setUp and tearDown methods to BeforeMethod and AfterMethod to ensure they run at the right times.

vteague committed:
Much cleaner IRV parsing without much impact on classes other than DominionCVRExportParser.java.

This now stores both Contest Choices and CVR choices as plain names (without parentheses).

vteague committed:
Deleting unused code.

vteague committed:
Deleting unused code.

vteague committed:
Deleting unused code.

vteague committed:
Deleting unused code.

vteague committed:
Checking for correct exception in tests.

michelleblom committed:
Added comment to ACVRUpload endpoint to mark point at which interpretation of IRV ballots needs to take place.

Made diluted margin in ComparisonAudit protected.

Added step to set diluted margin in IRVComparisonAudit.

Added tests to EstimateSampleSizesTest.

Added comment to EstimateSampleSizes endpoint.

michelleblom committed:
Merge pull request #56 from DemocracyDevelopers/IRVBallotsAsOrderedList

Irv ballots as ordered list

michelleblom committed:
Merge branch 'scratch' of https://github.com/DemocracyDevelopers/colorado-rla into scratch

vteague committed:
Tiny test cases designed for human redability and assertion / winner computation.

vteague committed:
Merge pull request #58 from DemocracyDevelopers/smallIRVTestCases

Tiny test cases designed for human redability and assertion / winner …

vteague committed:
Sample size estimation and assertion generation both responding and returning info, though still not joined up with the actual useful information.

vteague committed:
Sample size estimation and assertion generation both responding and returning info, though still not joined up with the actual useful information.

vteague committed:
Assertion generation endpoint connected to the right data and compiling.

vteague committed:
Assertion generation endpoint connected to the right data and compiling.

michelleblom committed:
Implemented CSV export to EstimateSampleSizes endpoint.

michelleblom committed:
Merge branch 'scratch' of https://github.com/DemocracyDevelopers/colorado-rla into scratch

vteague committed:
Assertion Generation endpoint working but not carefully tested.

vteague committed:
Assertion Generation endpoint working but not carefully tested.

vteague committed:
Some changes to address merge conflicts with scratch file rearrangement.

vteague committed:
Merge remote-tracking branch 'origin/feature/connector' into feature/connector

# Conflicts:
#	server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/GenerateAssertions.java
#	server/eclipse-project/src/main/java/us/freeandfair/corla/raire/response/Metadata.java

vteague committed:
Some changes to address merge conflicts with scratch file rearrangement.
